include/GafferScene/AttributeQuery.h

```diff
@@ -73,6 +73,15 @@ struct GAFFERSCENE_API AttributeQuery : Gaffer::ComputeNode
 	bool canSetup( const Gaffer::ValuePlug* plug ) const;
 	void setup( const Gaffer::ValuePlug* plug );
 
+	template< typename PlugType >
+	PlugType* defaultPlug();
+	template< typename PlugType >
+	const PlugType* defaultPlug() const;
+	template< typename PlugType >
+	PlugType* valuePlug();
+	template< typename PlugType >
+	const PlugType* valuePlug() const;
+
 	Gaffer::ValuePlug* defaultPlug();
 	const Gaffer::ValuePlug* defaultPlug() const;
 	Gaffer::ValuePlug* valuePlug();
```

Comment:
Perhaps I didn't explain this well enough before, but my intention was for a default for `PlugType` would mean we wouldn't need the original plug accessors at all. Maybe a diff explains better :

```
-       template< typename PlugType >
+       template<typename PlugType = Gaffer::ValuePlug>
        PlugType* defaultPlug();
-       template< typename PlugType >
+       template<typename PlugType = Gaffer::ValuePlug>
        const PlugType* defaultPlug() const;
-       template< typename PlugType >
+       template<typename PlugType = Gaffer::ValuePlug>
        PlugType* valuePlug();
-       template< typename PlugType >
+       template<typename PlugType = Gaffer::ValuePlug>
        const PlugType* valuePlug() const;
 
-       Gaffer::ValuePlug* defaultPlug();
-       const Gaffer::ValuePlug* defaultPlug() const;
-       Gaffer::ValuePlug* valuePlug();
-       const Gaffer::ValuePlug* valuePlug() const;
```